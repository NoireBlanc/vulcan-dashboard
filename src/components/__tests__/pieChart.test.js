import React from 'react';
import { act } from 'react-dom/test-utils';
import PieChart from '../pieChart';
import { formatPieData } from '../../utils/formatPieData.util';
import { payload } from '../../TestData/mockShipmentData';
import { mount, configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

global.ResizeObserver = jest.fn().mockImplementation(() => ({
    observe: jest.fn(),
    unobserve: jest.fn(),
    disconnect: jest.fn(),
}));

configure({ adapter: new Adapter() });
React.useLayoutEffect = React.useEffect;

describe('<PieChart />', () => {
    it('should render', () => {
        act(() => {
            let wrapper = mount(<PieChart pieData={formatPieData(payload[0].counts)} />);
            const { container } = wrapper;
            expect(container).toMatchSnapshot();
        });
    });
});
