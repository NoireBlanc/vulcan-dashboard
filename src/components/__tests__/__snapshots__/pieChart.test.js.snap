// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PieChart /> should render 1`] = `undefined`;
